src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -170,314 +153,55 @@ class FastSync(
-          updateTargetBlock(LastBlockValidationFailed)
-        } else {
+    def handleStateChange(state: SyncState)(f: SyncState => (FastSyncCommand, SyncState)): Unit ={
+      val (command, newState) = f(syncState)
+      syncState = newState
+      command match {
+        case ContinueSyncing(None) => processSyncing()
+        case ContinueSyncing(Some(blackListCommand)) =>
+          blacklist(blackListCommand.peer.id, blacklistDuration, blackListCommand.reason)
           processSyncing()
-        }
-      } else {
-        processSyncing()
-      }
-    }
-
-    private def handleBlockHeaders(peer: Peer, headers: Seq[BlockHeader]) = {
-      if (checkHeadersChain(headers)) {
-        processHeaders(peer, headers) match {
-          case ParentDifficultyNotFound(header) =>
-            log.debug("Parent difficulty not found for block {}, not processing rest of headers", header.number)
-            processSyncing()
-          case HeadersProcessingFinished =>
-            processSyncing()
-          case ImportedTargetBlock  =>
-            updateTargetBlock(ImportedLastBlock)
-          case ValidationFailed(header, peerToBlackList) =>
-            handleBlockValidationError(header, peerToBlackList, syncConfig.fastSyncBlockValidationN)
-        }
-      } else {
-        blacklist(peer.id, blacklistDuration, "error in block headers response")
-        processSyncing()
-      }
-    }
-
-    private def handleBlockBodies(peer: Peer, requestedHashes: Seq[ByteString], blockBodies: Seq[BlockBody]) = {
-      if (blockBodies.isEmpty) {
-        val reason = s"got empty block bodies response for known hashes: ${requestedHashes.map(h => Hex.toHexString(h.toArray[Byte]))}"
-        blacklist(peer.id, blacklistDuration, reason)
-        syncState = syncState.enqueueBlockBodies(requestedHashes)
-      } else {
-        validateBlocks(requestedHashes, blockBodies) match {
-          case BlockBodyValidationResult.Valid =>
-            insertBlocks(requestedHashes, blockBodies)
-          case BlockBodyValidationResult.Invalid =>
-            blacklist(peer.id, blacklistDuration, s"responded with block bodies not matching block headers, blacklisting for $blacklistDuration")
-            syncState = syncState.enqueueBlockBodies(requestedHashes)
-          case BlockBodyValidationResult.DbError =>
-            redownloadBlockchain()
-        }
-      }
-
-      processSyncing()
-    }
-
-    private def handleReceipts(peer: Peer, requestedHashes: Seq[ByteString], receipts: Seq[Seq[Receipt]]) = {
-      validateReceipts(requestedHashes, receipts) match {
-        case ReceiptsValidationResult.Valid(blockHashesWithReceipts) =>
-          blockHashesWithReceipts.foreach { case (hash, receiptsForBlock) =>
-            blockchain.save(hash, receiptsForBlock)
+        case InitTargetBlockUpdate(reason) =>
+          reason match {
+            case LastBlockValidationFailed(bl) => blacklist(bl.peer.id, blacklistDuration, bl.reason)
+            case _ =>
```

Comment:
probably missing something?

Reply:
actually i any other case i just wanted do nothing, i think it will be best to indicate it through  by using `()`